src/resolvers/get-model.ts

```diff
@@ -45,7 +45,15 @@ function getModelFromMakeDescription(make: Make, description: string): string |
       if (description.match(/(audi ((rs ?)|s)?q7|((^| )((rs ?)|s)?q7( |$)))/i)) {
         return Model[make].Q7
       }
-      if (description.match(/(audi ((rs ?)|s)?q8|((^| )((rs ?)|s)?q8( |$)))/i)) {
+
+      const matchesQ8 = description.match(/(audi ((rs ?)|s)?q8|((^| )((rs ?)|s)?q8( |$)))/i)
+      const matchesEtron = description.match(/(audi e-tron|((^| )e-tron( |$)))/i)
+
+      if (matchesQ8 && matchesEtron) {
+        return Model[make].Q8_ETRON
+      }
+
+      if (matchesQ8) {
```

Comment:
```suggestion
      const matchesQ8 = description.match(/(audi ((rs ?)|s)?q8|((^| )((rs ?)|s)?q8( |$)))/i)
      const matchesEtron = description.match(/(audi e-tron|((^| )e-tron( |$)))/i)

      if (matchesQ8 && matchesEtron) {
        return Model[make].Q8_ETRON
      }

      if (description.match(/(audi ((rs ?)|s)?q8|((^| )((rs ?)|s)?q8( |$)))/i)) {
        if (description.match(/((^| )e-tron( |$))/i)) {
          return Model[make].Q8_ETRON
        }
```

Reply:
Can you elaborate on these suggestions?

Reply:
Basically to make the hierarchy clearer (that the Q8 e-tron is a "submatch" to Q8) and to keep the regexes as simple as possible (`audi e-tron` is unnecessary to look for with Q8 e-tron as it will never contain that string).

Reply:
But the branch on line 61 is already covered by the condition on line 55 no? Should the `matchesQ8` and `matchesEtron` parts be removed then? That would make more sense.

Reply:
> audi e-tron is unnecessary to look for with Q8 e-tron as it will never contain that string

I opted for less redundancy here although this is correct.